### DISCUSSION TITLE
Tailwind won't load some attributes with Django

### DISCUSSION THREAD

**Discussion by vincentrandon** [tailwindlabs/tailwindcss#8837] [Help]
I have set up TailwindCSS v.3 on my Django website thanks to `django-tailwind`. 
On a regular basis, for example, some colors or some attributes won't load on my website and I have looking for days, I cannot find the reasonwhy.
Let's take the following button:
```
<button type="button" id="custom-button" class="bg-green-600 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded-full"> Upload </button>
```
The bg-color and the hover won't display:
<img width="512" alt="Capture d’écran 2022-07-12 à 17 38 55" src="https://user-images.githubusercontent.com/103646578/178531854-2a631b2b-ae1c-4507-b149-af4682325970.png">

If I change this button to:
```
<button type="button" id="custom-button" class="bg-[#68BA9E] hover:bg-blue-700 text-white font-bold py-2 px-4 rounded-full"> Upload </button>
```
The bg-color will display:
<img width="516" alt="Capture d’écran 2022-07-12 à 17 40 41" src="https://user-images.githubusercontent.com/103646578/178532225-ca12dac0-f8b9-4ec7-b4ea-b19c4286d410.png">

I always clear my cache, I already tried to restart my server, nothing happens... Could you help me with that? 

Thanks!

**Comment by skyforest** [comment]
@vincentrandon did you ever figure this out?

**Reply by skyforest** [reply]
For my own issue, I was seeing a similar issue when rendering a crispy form (django+tailwind+crispyforms). If I used a form helper to render the button with tailwind classes, it would just show white text like in your first example. But, when i went and hardcoded the button with the same tailwind classes in the html template, django-tailwind would then and only then apply the discovered styling to the form-helper button as well.

So, in my case, the auto-reload and application of tailwind styles needs to be rendered in the actual html template for django-tailwind to do it's thing. Otherwise, any internally rendered element (like a button rendered from the django-crispyform helper) won't receive the styling.

**Reply by vincentrandon** [reply]
I did figure it out. 
In my case, I ran `python manage.py collectstatic` and it created the following directory: `static > css > dist > styles.css`. 
When I first setup `django-tailwind`, a `theme` folder was created with also a `theme > static > css > dist > styles.css`. I ended up with the two same files and it seems logical to say that Django was not taking right one.
The solution for me was to rename `static > css > dist > styles.css` to `static > css_ > dist > styles.css` and then it worked.
